doc/modelreference.md

```diff
@@ -0,0 +1,225 @@
+# GBDXM Model Support
+
+## General Parameters
+General parameters are ones that are used during processing regardless of the
+type of model.  They may be used to determine the appropriate input imagery.
+```
+  -l [ --labels ] PATH                  Labels file name.
+  --label-names LABEL1 [LABEL2 ...]     A list of label names.
+  -w [ --model-size ] WIDTH [HEIGHT]    Classifier model size. Model parameters
+                                        will override this if present.
+  -c [ --color-mode ] MODE              Color mode. Model parameters will 
+                                        override this if present. Must be one 
+                                        of the following: grayscale, rgb, 
+                                        multiband
+```
+
+#### Color Mode
+ - `grayscale` - The model and data must be single band.
+ - `rgb` - The model and data must be 3 band, models are expected to be a 
+ - `multiband` - The model and data may be any number of bands, but it must match
+ 
+#### Model Size
+ 
+ 
+## Caffe
+
+Currently supported is the [NVIDIA branch of Caffe at version 15](https://github.com/NVIDIA/caffe/tree/caffe-0.15).  Except for where noted below,
+only the stock c++  layers are supported.  Caffe was originally developed by the
+[Berkeley Artificial Intelligence Research](http://caffe.berkeleyvision.org/) 
+(BAIR) Lab (formerly known as the Berkeley Vision and Learning Center (BVLC)).
+
+### Options
+
+```
+-C [ --category ] CATEGORY            Model category (optional). The 
+                                      supported categories are: classifier, 
+                                      detectnet, segmentation.
+--caffe NAME VALUE [NAME VALUE ...]   Set multiple Caffe options.
+--caffe-model PATH                    Caffe model topology file name.
+--caffe-trained PATH                  Caffe model weights file name.
+--caffe-mean PATH                     Caffe model mean values file name 
+                                      (optional).
+--caffe-output-layer VALUE            Output layer name (optional).
+```
+
+### General Notes
+
+### Classifier Category
+
+A classifier accepts a patch of the input image and determines the likelihood that
+it is each of the known labels.
+
+Input layer:
+ - _Input image_ of size {`batch size`, `number of channels`, `image height`, 
+ `image width`}.
+ 
+Output layer:
+ - _Data_ of size {`batch size`, `number of labels`}.  This must be in the top
+ of the network and have a "Softmax" type.  Each element refers to the 
+ likelihood that that label describes the image.  The labels must be ordered
+ in the same order as they are in the model metadata.
+ 
+Notes:
+ - The "output-layer" parameter is ignored.
```

Comment:
What do you mean by that? The output layer parameter is not ignored. However, it is not required if it's the only "Softmax" top layer.

Reply:
Turns out that's not the case, though it can be considered a bug that it's ignored like that.